app/screens/create-match.js

```diff
@@ -51,11 +61,14 @@ export class CreateMatch extends React.PureComponent {
         image
       }
     };
+    if (this.state.switched) {
+      match = { ...match, totalPlayers: this.state.totalPlayers }
+    };
     return match;
   }
 
   handleClick(event) {
-    this.props.createMatch(this.match, this.onSuccess);
+    this.props.createMatch(this.getMatch(), this.onSuccess);
```

Comment:
Hacer la function que retorna createMatch retorne una Promise. entonces podés hacer:

`createMatch(this.getMatch()).then(onSuccess);`

El problema que tiene pasale onSuccess como callback a createMatch, es que el `catch` de `matchService.create` puede catchear errores generador por onSucess.

Reply:
el history.push que hace onSucess se debería de hacer en una async action

http://redux.js.org/docs/advanced/AsyncActions.html

Reply:
Buenas, estamos viendo esto con Juan.
this.props.createMatch es una async action, estaría bien cambiar la firma de 
createMatch sacando el callback, y que llame una async action encadenada? (la que ahora seria on success), o capaz mejor, dado que ya estamos en una async action, redirigir desde ahí (con react-router-redux)?

O sea, la firma final de creatcreateeMatch quedaría:
createMatch: Match -> Dispatch -> Ø 
y como efectos colaterales, estaría el despache de 3 acciones puras (indicando inicio, fin con éxito(en el then), o fracaso(en el catch)) y una impura/asincronica que seria el onSucces, capaz cambiando el nombre por redirect o algo así. y que esta ultima solo haga dispatch(push(...)).
como dije en el primer párrafo, esta ultima se podría obviar, pero capaz sea útil tenerla para acciones futuras.

que opina?

Reply:
Buenas, podemos cerrar esto y dejarlo como known issue para el sprint siguiente? gracais!